### PR TITLE
(coffee to jsx) prep for new datatypes input mapping form

### PR DESCRIPTION
**Pull request by pivnicek**
Prep for #1677 

Proposed changes:

- convert InputMappingRowSnowflakeEditor to jsx

### CODE REVIEW

**Comment by ondrejhlavacek** [comment]
sweet, will get to it today!

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      let mutation = mapping.set('columns', newValue);
+      if (newValue.count()) {
+        let columns = mutation.get('columns').toJS();
+        if (!_.contains(columns, mutation.get('whereColumn'))) {
+          mutation = mutation.set('whereColumn', '');
+          mutation = mutation.set('whereValues', Immutable.List());
+          mutation = mutation.set('whereOperator', 'eq');
+        }
+        let datatypes = _.pick(mutation.get('datatypes').toJS(), columns);
+        mutation = mutation.set('datatypes', Immutable.fromJS(datatypes || Immutable.Map()));
```
Comment (ujovlado) [inline src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: missing return

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    tables: React.PropTypes.object.isRequired,
+    onChange: React.PropTypes.func.isRequired,
+    disabled: React.PropTypes.bool.isRequired,
+    initialShowDetails: React.PropTypes.bool.isRequired,
+    isDestinationDuplicate: React.PropTypes.bool.isRequired
+  },
+
+  _handleChangeSource(value) {
+    // use only table name from the table identifier
+    var destination;
```
Comment (ujovlado) [inline src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: this should be simplified: `const destination = value ? ... : ''`

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+import Immutable from 'immutable';
+import {Input} from './../../../../../react/common/KbcBootstrap';
+
+import Select from '../../../../../react/common/Select';
+import SapiTableSelector from '../../../../components/react/components/SapiTableSelector';
+import SnowflakeDataTypesContainer from './input/SnowflakeDataTypesContainer';
+import ChangedSinceInput from '../../../../../react/common/ChangedSinceInput';
+import {PanelWithDetails} from '@keboola/indigo-ui';
+
+module.exports = React.createClass({
```
Comment (ujovlado) [inline src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: use ES export (`export default`)

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    return _.map(columns, (column) => {
+      return {
+        label: column,
+        value: column
+      };
+    });
+  },
+
+  _getFilteredColumnsOptions() {
+    var columns;
```
Comment (ujovlado) [inline src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: const/let

**Comment by pivnicek** [comment]
Thanks Ujo!  Updating now

**Comment by ondrejhlavacek** [comment]
this error message
image
became
image

possibly related to the exports?

the bug is in all input mappings, appears only once per refresh and is triggered by opening the new input mapping modal

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx (hunk 1 above)
Comment (pivnicek) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: ?  nechapu

**Comment by ondrejhlavacek** [comment]
apart from that everything looks working

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx (hunk 1 above)
Comment (ujovlado) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: there is missing `return` statement

**Comment by pivnicek** [comment]
@ondrejhlavacek yeah, I had a look at that briefly.  I think it's in `modals/inputMapping.jsx`

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx (hunk 1 above)
Comment (ujovlado) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: screenshot_2018-05-28_13-59-52 ... you're in `map` ... this is not coffee where last statement is also returned

Comment (pivnicek) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: ahhhhh  duh!  dik

**Comment by ondrejhlavacek** [comment]
after changing to es exports it's back to 

image

so all good!

**Comment by ondrejhlavacek** [comment]
@pivnicek no need to fix it, i was just worried about the error message leading nowhere :-)

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+                </div>
+              </div>
+              <div className="form-group">
+                <label className="col-xs-2 control-label">Data types</label>
+                <div className="col-xs-10">
+                  <SnowflakeDataTypesContainer
+                    value={this.props.value.get('datatypes', Immutable.Map())}
+                    disabled={this.props.disabled || !this.props.value.get('source')}
+                    onChange={this._handleChangeDataTypes}
+                    columnsOptions={this._getColumnsOptions()}
```
Comment (ujovlado) [inline src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: @pivnicek is there a reason why `_getFilteredColumnsOptions` was changed to `_getColumnsOptions`?

Comment (pivnicek) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: @ujovlado no!  thank you for catching that, sorry!

**Comment by ujovlado** [comment]
@pivnicek 

fast way to chcek if `coffee` to `jsx` conversion produces same component:

Create `InputMappingRowSnowflakeEditor.test.js` (this case) before rewrite with contents:

```js
import React from 'react';
import Immutable from 'immutable';
import InputMappingRowSnowflakeEditor from './InputMappingRowSnowflakeEditor';

describe('<InputMappingRowSnowflakeEditor />', function() {
  it('should render InputMappingRowSnowflakeEditor - 1', function() {
    shallowSnapshot(
      <InputMappingRowSnowflakeEditor
        value={Immutable.fromJS({
          source: 'in.c-ex.src',
          destination: 'dest',
        })}
        tables={Immutable.fromJS([

        ])}
        onChange={() => {}}
        disabled={false}
        initialShowDetails={true}
        isDestinationDuplicate={false}
      />
    );
  });

 // write as many cases (combination of props) as possible
});
```

Run `yarn tdd`, then select `u` (read instructions) to create new snapshots.

Rewrite to `jsx`.

Check if snapshot matches.

File: src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx (hunk 5 above)
Comment (ujovlado) [reply src/scripts/modules/transformations/react/components/mapping/InputMappingRowSnowflakeEditor.jsx]: Fixed. I'll merge it

**Pull request merged**